opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -226,7 +227,9 @@ def add_link(
         attributes: types.Attributes = None,
     ) -> None:
         if attributes is None:
-            attributes = Span.empty_attributes
+            attributes = (
+                Span.empty_attributes
+            )  # TODO: empty_attributes is not a Dict. Use Mapping?
```

Comment:
That sounds like the right solution to me, why not do it in this PR?

Reply:
The PR is currently marked WIP (maybe I should've made it a Draft PR). Since there are so many "problems" I'm not sure if this is something that is really worthwhile.